STYLE_GUIDE.md

```diff
@@ -0,0 +1,179 @@
+# Style Guide
+
+In datarockets we enforce a community [Ruby Style Guide](https://github.com/rubocop-hq/ruby-style-guide).
+
+This is a small list of differences which we have when compared with community style guide:
+
+## Bundler
+
+* A Gem's requirements should be listed only once in a Gemfile
+
+* Gems should be alphabetically sorted within groups. Also you can use a line comment as a group separator.
+
+## Style
+
+* Limit lines to 80 characters.
+
+* Prefer double-quotes unless your string literal contains " or escape characters you want to suppress.
+
+* If elements of a hash literal span more than one line we're aligning them by keys
+
+
+```ruby
+# bad
+{
+  foo: bar,
+   ba: baz
+}
+{
+  foo: bar,
+  ba:  baz
+}
+
+# good
+{
+  foo: bar,
+  ba: baz
+}
+```
+
+* There is one exception in the rule above when the hash literal is passed as method arguments.
+In this case we're just using an indentation level.
+
+```ruby
+# bad
+do_something(foo: 1,
+             bar: 2)
+
+# good
+do_something(foo: 1,
+  bar: 2)
+```
+
+* The parameters on a multi-line method call or definition are aligning by an indentation level.
+
+```ruby
+# good
```

Comment:
in this particular case I prefer to use
```
def foo(bar, baz)
   "#{bar} #{baz}"
end
```

Reply:
@DDKatch heh, me too. But what will you do if you need to use multi-line method call (in any cases 🤔 )? E.g

```ruby
def foo(bar1, baz1, bar12, baz12, bar11, baz13, bar13, baz11, bar14, baz14, bar15, baz15, bar16, baz16, bar17, baz17, bar18, baz18, bar19, baz19)
   "#{bar} #{baz}"
end
```

Reply:
@roman-dubrovsky we code in ruby, not C/C++ why the hell we will write the code in this creepy style?

Reply:
@DDKatch cause we love services which can get many arguments (just change all arguments to keyword parameters)

Reply:
and long well-described names

Reply:
There shouldn't be more than 2-3 arguments for one service or method, otherwise, service/method knows too much. Thoughts?

Reply:
@AleksSenkou I don't agree. Yes, probably for services we shouldn't pass a big list of arguments (and can't use this rule every time) and should use something like DTO pattern. But here, we have two issues:

1) For initializing DTO we usually should pass more than 2-3 object, e.g.

```ruby
class Post::CreateArguments
  def initialize(title:, note:, user:, 
    object:, tags:, links:)
end
``` 

2) Sometimes we should pass more than two arguments for services as well. E.g. case from our project. We have a service (with post some info to the external service) which gets these arguments:

1) client - wrapper, which we're using for sending requests to external API
2) contact - a data which we're sending to external service
3) history (optional) - some action which we made via contact object. We use it for getting more and specific data for sending
4) owner_id - External User UID which should be the owner of the new (or updated) entity
5) create_deal (optional, by default - true) - a boolean flag which checks should we create an additional entity (deal) in the external service
6) deal_config (optional) - DTO object where user can prepare some data for the created deal object

Reply:
Also, there are some restrictions with rubocop rules (I used examples from rubocop doc) so we can't restrict some examples

Anyway, I can change it like this and use comment better for some cases

```ruby
# good

def foo(bar,
  baz)
  123
end

# better

def foo(
  bar,
  baz
)
  123
end
```

Thoughts?

Reply:
> ```ruby
> # good
> 
> def foo(bar,
>   baz)
>   123
> end
> 
> # better
> 
> def foo(
>   bar,
>   baz
> )
>   123
> end
> ```
To be honest both of them are hard to read and understand

Reply:
ups, my bad (looks like thought about other thing)

if I change it to the example below, will it be better? 


```ruby
def foo(bar, baz, kek,
  lol)
  123
end
```

BTW this is an example for a rule below so pls feel free to suggest another example 

https://rubocop.readthedocs.io/en/latest/cops_layout/#layoutalignparameters

Reply:
cc @TheBlackArroVV

Reply:
I mostly like  this one
```ruby
def foo(bar,
            baz)
  123
end
```

Reply:
I don't. So feel free to add it to the separate PR (check label) so that we can discuss it separately.

P.S. in this PR I only opens to change rules and example w/o breaking and changing rubocop config. This issues I prefer to discuss separately and change after resealing the first version of the gem. Does it make sense?

Reply:
Sure

Reply:
@TheBlackArroVV one comment. Not PR but the separate issue

Reply:
@roman-dubrovsky agreed 👍 

I just trying to split services not to use a huge amount of args. I still believe that a situation on your project can be refactored to use fewer arguments, but I can't suggest anything without digging into the codebase.

Reply:
> I just trying to split services not to use a huge amount of args.

Reply:
> I just trying to split services not to use a huge amount of args. 

but in this case, you still have a service which collects all arguments and calls smaller services, don't you?

Reply:
@roman-dubrovsky that's true